### PR TITLE
rpc: Make gettxoutsetinfo/GetUTXOStats interruptible

### LINKED ISSUE #13217: [GUI] QT shutdown is blocked if called before gettxoutsetinfo is finished
Steps to reproduce:
1) Open QT console.
2) Run gettxoutsetinfo.
3) Exit the debug window (no blocking) -> exit the program (will block).

This is what is shown on Windows (10) for several minutes:
[https://i.imgur.com/AyexgkC.png](https://i.imgur.com/AyexgkC.png)

### PR DESCRIPTION
**Pull request by maflcko**
Make it interruptible, so that shutdown doesn't block for up to one hour.

Fixes (partially) #13217

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Can be tested with the cli:

```
$ ./src/bitcoin-cli -datadir=/run/media/marco/b_re/TEMP_del//bt gettxoutsetinfo 
error code: -9
error message:
Shutting down

**Comment by Empact** [comment]
Code Review ACK https://github.com/bitcoin/bitcoin/pull/19056/commits/fa756928c3f455943086051c5fe1d5bb09962248

**Review by promag — COMMENTED** [review]
Concept ACK.

**Comment by hebasto** [comment]
@MarcoFalke mind looking into #17659 ?

**Comment by hebasto** [comment]
Concept ACK.

File: src/node/coinstats.h [hunk 1]
```diff
@@ -29,6 +30,6 @@ struct CCoinsStats
 };
 
 //! Calculate statistics about the unspent transaction output set
-bool GetUTXOStats(CCoinsView* view, CCoinsStats& stats);
+bool GetUTXOStats(CCoinsView* view, CCoinsStats& stats, const std::function<void()>& interruption_point = {});
```
Comment (hebasto) [inline src/node/coinstats.h:33]: Is default ` = {}` really needed?

**Review by hebasto — COMMENTED** [review]
While touching the body of `FindScriptPubKey()` is it a good chance to make it `static`?

**Comment by laanwj** [comment]
Code review ACK fa756928c3f455943086051c5fe1d5bb09962248

File: src/node/coinstats.h (hunk 1 above)
Comment (MarcoFalke) [reply src/node/coinstats.h:33]: Yes, otherwise all callers are forced to specify an interruption op

Comment (MarcoFalke) [reply src/node/coinstats.h:33]: See for example:

```
$ git grep --extended-regexp  'GetUTXOStats\([^,]+,[^,]+\)'
src/test/fuzz/coins_view.cpp:                (void)GetUTXOStats(&coins_view_cache, stats);

**Pull request merged**